### PR TITLE
Implementation of lists: VehicleOccupancies, VehicleGtfsRT and VehicleJourney

### PR DESCRIPTION
**Pull request by julyto**
Implementation for list management:
- VehicleOccupancies,  (for the test, this list is emptied every 10 minutes)
- VehicleGtfsRT
- VehicleJourney (for the test, the elements older than one hour are deleted)

in Context.go I added the ``` VehicleOccupanciesGtfsRt map [int]*VehicleOccupancy```, after that it will be the same for Oditi

Add date field to VehicleJourney structure and update tests

### CODE REVIEW

File: internal/vehicleoccupancies/context.go [hunk 1]
```diff
@@ -16,6 +16,7 @@ import (
 ------------------------------------------------------------- */
 type VehicleOccupanciesContext struct {
 	VehicleOccupancies           *map[int]VehicleOccupancy
```
Comment (benoit-bst) [inline internal/vehicleoccupancies/context.go:18]: So we keep this map ? `VehicleOccupancies` is not replace by `VehicleOccupanciesGtfsRt` ?

**Review by benoit-bst — APPROVED** [review]

File: internal/vehicleoccupancies/context.go (hunk 1 above)
Comment (julyto) [reply internal/vehicleoccupancies/context.go:18]: After refactoring `vehicleOccupancies Oditi` this map will be type map[int]VehicleOccuanpy (as `VehicleOccupanciesGtfsRt`)

**Review by benoit-bst — APPROVED** [review]

File: internal/vehicleoccupancies/contextGtfsRt.go [hunk 2]
```diff
@@ -117,79 +176,114 @@ func (d *VehicleOccupanciesGtfsRtContext) LoadDataExternalSource(uri url.URL, to
...
+
+			// add in vehicle occupancy list
+			newVehicleOccupancy := createOccupanciesFromDataSource(*vj, vehGtfsRT)
+			if newVehicleOccupancy != nil {
+				context.AddVehicleOccupancy(newVehicleOccupancy)
+			}
 		} else {
-			// Get status Last load from Navitia and check if data loaded recently
-			vj, _ := GetVehicleJourney("id_code", navitiaURI, navitiaToken, connectionTimeout)
+			vj = context.vehiclesJourney[vehGtfsRT.Trip]
```
Comment (kadhikari) [inline internal/vehicleoccupancies/contextGtfsRt.go:249]: if gtfs-rt vehicle exists in map of vehicle occupancies does not mean that vehicleJourney exits in 'context.vehiclesJourney' as two lists are cleaned with different durations.

File: internal/vehicleoccupancies/contextGtfsRt.go [hunk 3]
```diff
@@ -18,57 +20,86 @@ import (
...
+var start = time.Now() // TODO: Just for test, delete to release
+
+func (d *VehicleOccupanciesGtfsRtContext) GetVehicleJourneys() (vehicleJourneys map[string]*VehicleJourney) {
+	d.mutex.Lock()
+	defer d.mutex.Unlock()
+	return d.vehiclesJourney
 }
 
 func (d *VehicleOccupanciesGtfsRtContext) GetVehiclesGtfsRts() (vehiclesGtfsRts []VehicleGtfsRt) {
+	d.mutex.Lock()
```
[the remark this answers is not in the document]
Comment (julyto) [reply internal/vehicleoccupancies/contextGtfsRt.go:39]: yes you are right I will modify the geter

File: internal/vehicleoccupancies/contextGtfsRt.go (hunk 2 above)
Comment (julyto) [reply internal/vehicleoccupancies/contextGtfsRt.go:249]: Normally we should not have the case, the VJourney map is cleaned  every day and the VOccupancies map every x hours. But to make sure I don't have this case, I'll move the GetVehicleJourney to the start of the loop.

**Review by kadhikari — APPROVED** [review]

**Pull request merged**